core/trino-main/src/main/java/io/trino/operator/aggregation/state/LongDecimalWithOverflowStateSerializer.java

```diff
@@ -52,19 +62,19 @@ public void serialize(LongDecimalWithOverflowState state, BlockBuilder out)
     public void deserialize(Block block, int index, LongDecimalWithOverflowState state)
     {
         if (!block.isNull(index)) {
-            Slice slice = VARBINARY.getSlice(block, index);
-            if (slice.length() != SERIALIZED_SIZE) {
-                throw new IllegalStateException("Unexpected serialized state size: " + slice.length());
+            Block row = block.getObject(index, Block.class);
```

Comment:
you should directly use `getRawFieldBlocks` combined with `io.trino.spi.block.RowBlock#getFieldBlockOffsets`

Using `block.getObject(index, Block.class);` is pretty expensive as it creates block hierarchy every time